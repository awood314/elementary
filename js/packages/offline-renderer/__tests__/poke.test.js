import OfflineRenderer from '../index';
import { el } from '@elemaudio/core';


test.only('poke node', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 0,
    numOutputChannels: 1,
  });

  // Graph
  core.render(el.poke());

  // Ten blocks of data
  let inps = [];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Now we should be able to poke the graph and see the impulse on the next block
  core.poke();

  inps = [];
  outs = [new Float32Array(32)];

  core.process(inps, outs);
  expect(outs[0]).toMatchSnapshot();
});
