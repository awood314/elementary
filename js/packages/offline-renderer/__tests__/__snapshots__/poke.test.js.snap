// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poke node 1`] = `
Float32Array [
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
]
`;
